nixos/modules/programs/steam.nix

```diff
@@ -133,6 +150,12 @@ in {
 
     hardware.steam-hardware.enable = true;
 
+    # Append the extra compatibility packages to whatever else the env variable was populated with.
+    # For more information see https://github.com/ValveSoftware/steam-for-linux/issues/6310#issuecomment-511630263
+    environment.sessionVariables = mkIf (cfg.extraCompatPackages != []) {
+      STEAM_EXTRA_COMPAT_TOOLS_PATHS = strings.makeBinPath cfg.extraCompatPackages;
+    };
+
```

Comment:
I'd prefer to scope this variable to just steam via its wrapper.

Reply:
we could do something like this https://github.com/Shawn8901/nixpkgs/commit/3f09abfa73b9d8fcef55be83e5d24afb02df31d9?

Reply:
Yup, that's what I meant.

Reply:
@romatthe  do you want to change the PR accordingly or should i create a pr that replaces this?

Reply:
i did create https://github.com/NixOS/nixpkgs/pull/293564  as a follow up.